deepchem/splits/tests/test_splitter.py

```diff
@@ -1,72 +1,147 @@
+                frac_train=0.8, frac_valid=0.1, frac_test=0.1)
+        assert len(train_data) == 8
+        assert len(valid_data) == 1
+        assert len(test_data) == 1
+
+    def test_multitask_random_split(self):
+        """
+        Test multitask RandomSplitter class.
+        """
+        multitask_dataset = self.load_multitask_data()
+        random_splitter = RandomSplitter()
+        train_data, valid_data, test_data = \
+            random_splitter.train_valid_test_split(
+                multitask_dataset,
+                self.train_dir, self.valid_dir, self.test_dir,
+                frac_train=0.8, frac_valid=0.1, frac_test=0.1)
+        assert len(train_data) == 8
+        assert len(valid_data) == 1
+        assert len(test_data) == 1
+
+    def test_multitask_scaffold_split(self):
+        """
+        Test multitask ScaffoldSplitter class.
+        """
+        multitask_dataset = self.load_multitask_data()
+        scaffold_splitter = ScaffoldSplitter()
+        train_data, valid_data, test_data = \
+            scaffold_splitter.train_valid_test_split(
+                multitask_dataset,
+                self.train_dir, self.valid_dir, self.test_dir,
+                frac_train=0.8, frac_valid=0.1, frac_test=0.1)
+        assert len(train_data) == 8
+        assert len(valid_data) == 1
+        assert len(test_data) == 1
+
+    def test_stratified_multitask_split(self):
+        """
+        Test multitask StratifiedSplitter class
+        """
+        #ensure sparse dataset is actually sparse
+
+        sparse_dataset = self.load_sparse_multitask_dataset()
+
+        X, y, w, ids = sparse_dataset.to_numpy()
+
+        """
+        sparsity is determined by number of w weights that are 0 for a given task
+        structure of w np array is such that each row corresponds to a sample -- e.g., analyze third column for third
+        sparse task
+        """
+        frac_train = 0.5
+        cutoff = int(frac_train * w.shape[0])
+        w = w[:cutoff, :]
+        sparse_flag = False
 
-  def test_singletask_scaffold_split(self):
-    """Test singletask ScaffoldSplitter class."""
-    solubility_dataset = self.load_solubility_data()
-    scaffold_splitter = ScaffoldSplitter()
-    train_data, valid_data, test_data = \
-        scaffold_splitter.train_valid_test_split(
-            solubility_dataset,
-            self.train_dir, self.valid_dir, self.test_dir,
-            frac_train=0.8, frac_valid=0.1, frac_test=0.1)
-    assert len(train_data) == 8
-    assert len(valid_data) == 1
-    assert len(test_data) == 1
+        colIndex = 0
+        for col in w.T:
+            if not np.any(col): #check to see if any columns are all zero
+                sparse_flag = True
+                break
+            colIndex+=1
+        if not sparse_flag:
+            print("Test dataset isn't sparse -- test failed")
+        else:
+            print("Column %d is sparse -- expected" % colIndex)
+        assert sparse_flag is True
```

Comment:
You can just say `assert sparse_flag`. No need for an explicit boolean comparison.